src/core/interpreter/index.ts

```diff
@@ -1140,7 +1140,34 @@ export class Interpreter implements Expr.Visitor<BrsType>, Stmt.Visitor<BrsType>
                         signature: signature,
                     });
                     try {
-                        const returnValue = callee.call(this, ...args);
+                        let returnValue = callee.call(this, ...args);
+                        if (
+                            signature.returns !== ValueKind.Void &&
+                            signature.returns !== ValueKind.Dynamic &&
+                            returnValue.kind === ValueKind.Invalid
```

Comment:
how does this distinguish from real `return invalid` ? 
```
Brightscript Debugger> ? (function() as object: return invalid: end function)()
<Component: roInvalid>
```

Reply:
The real `return` is a Javascript exception (we use `throw` to return from functions) and it will reach other part of the interpreter code, this particular place is the one that the call is returned without hitting the `return` statement.